### ISSUE TITLE
[BUG] Android button's text in Popup not aligned to center

### ISSUE THREAD

**Issue by czmirek** [CommunityToolkit/Maui#1719]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Button text is not centered when inside a popup in the reproduction repository here: https://github.com/czmirek/MauiButtonAlignBug

### Expected Behavior

The button text is aligned to center correctly.

### Steps To Reproduce

- https://github.com/czmirek/MauiButtonAlignBug
- Run in debug in Android Simulator --- Android 12.1, API 32
- Click on "Show popup"

<img width="362" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/18615545/cbd02baf-813b-443c-a1f4-1cffdb75ce0c">

- Observe button text not centered

<img width="350" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/18615545/d93f17a7-ad52-43ce-92a7-45ca86e0fd18">

### Link to public reproduction project repository

https://github.com/czmirek/MauiButtonAlignBug

### Environment

```markdown
- .NET MAUI CommunityToolkit: 7.0.1
- .NET MAUI: 8.0.7
- OS: Android 12.1, API 32
- Happens in both Debug and Release
```

### Anything else?

- Go to TestPopup.xaml and remove the style from the `<Button>`
- It is centered now
<img width="330" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/18615545/59dbea64-ae73-4f01-bab6-408bd31f0d6d">

But as you can see in `App.xaml` there is nothing in the styles that should affect how the text inside the button is aligned.
```xml
        <ResourceDictionary>
            <Color x:Key="PrimaryColor">#FFBB00</Color>
            <Color x:Key="TextColor">#000000</Color>
            <Style TargetType="Button" x:Key="BaseButton">
                <Setter Property="CornerRadius" Value="{OnPlatform 25}" />
                <Setter Property="Padding" Value="20,10,20,10" />
                <Setter Property="FontSize" Value="16" />
                <Setter Property="HeightRequest" Value="48" />
            </Style>
            <Style TargetType="Button" x:Key="PrimaryButton" BasedOn="{StaticResource BaseButton}">
                <Setter Property="BackgroundColor" Value="{StaticResource PrimaryColor}" />
                <Setter Property="TextColor" Value="{StaticResource TextColor}" />
            </Style>
        </ResourceDictionary>
```

**Comment by blakepulman** [comment]
I am also seeing this issue, new to .NET 8, only on Android.  
I have been able to recreate this problem on .NET 8 (versions 8.0.3 and 8.0.7) with CommunityToolkit versions 5.3.0, 6.1.0 and 7.0.1. 
Switching back to .NET 7 with versions 5.3.0 and 6.1.0 fixes the problem (I am unable to do this with 7.0.1 as it is incompatible with .NET 8). 

The behaviour I observe is that even without a style applied (except for the default in Styles.xaml), the text is still not centred. 

This behaviour does not occur outside Popups (i.e. a duplicate of the popup button in a ContentView / ContentPage will have its text centre-aligned).

**Comment by vhugogarcia** [comment]
Thanks @czmirek for posting the issue. We ran some tests on Android 9, 12.1 and 14, and we can confirm the issue has been fixed on the latest of the MAUI Community Toolkit 9.0.2 and .NET MAUI 8.0.70.

I'm closing this issue, we can re-open it if the issue appears again.

<img width="311" alt="Screenshot 2024-07-24 at 10 01 06 a m" src="https://github.com/user-attachments/assets/cf8a8647-3ab5-49bd-82d2-512e88414f11">
